cluster-autoscaler/cluster_autoscaler.go

```diff
@@ -53,27 +49,55 @@ func main() {
 	}
 
 	kubeClient := kube_client.NewOrDie(kubeConfig)
-
-	// watch unscheduled pods
-	selector := fields.ParseSelectorOrDie("spec.nodeName==" + "" + ",status.phase!=" +
-		string(kube_api.PodSucceeded) + ",status.phase!=" + string(kube_api.PodFailed))
-	podListWatch := cache.NewListWatchFromClient(kubeClient, "pods", kube_api.NamespaceAll, selector)
-	podListener := &cache.StoreToPodLister{Store: cache.NewStore(cache.MetaNamespaceKeyFunc)}
-	podReflector := cache.NewReflector(podListWatch, &kube_api.Pod{}, podListener.Store, time.Hour)
-	podReflector.Run()
+	unscheduledPodLister := NewUnscheduledPodLister(kubeClient)
+	nodeLister := NewNodeLister(kubeClient)
 
 	for {
 		select {
 		case <-time.After(time.Minute):
 			{
-				pods, err := podListener.List(labels.Everything())
+				pods, err := unscheduledPodLister.List()
 				if err != nil {
 					glog.Errorf("Failed to list pods: %v", err)
-					break
+					continue
+				}
+				if len(pods) == 0 {
+					glog.V(1).Info("No unscheduled pods")
+					continue
 				}
 
 				for _, pod := range pods {
-					glog.Infof("Pod %s/%s is not scheduled", pod.Namespace, pod.Name)
+					glog.V(1).Infof("Pod %s/%s is not scheduled", pod.Namespace, pod.Name)
+				}
+
+				nodes, err := nodeLister.List()
+				if err != nil {
+					glog.Errorf("Failed to list nodes: %v", err)
+					continue
+				}
+				if len(nodes.Items) == 0 {
+					glog.Errorf("No nodes in the cluster")
+					continue
+				}
+
+				// TODO: Checking if all nodes are present.
+
+				// Checks if scheduler tried to schedule the pods after thew newest node was added.
+				newestNode := GetNewestNode(nodes.Items)
+				if newestNode == nil {
+					glog.Errorf("No newest node")
+					continue
+				}
+				oldesSchedulingTrial := GetOldestFailedSchedulingTrail(pods)
+				if oldesSchedulingTrial == nil {
+					glog.Errorf("No oldest unschedueled trial: %v", err)
+					continue
+				}
+
+				if newestNode.CreationTimestamp.After(oldesSchedulingTrial.Add(-1 * time.Minute)) {
```

Comment:
Let's add a TODO that we should find a better way to verify that scheduler took into account newest node.